### DISCUSSION TITLE
65“ TV What Hardware i Need?

### DISCUSSION THREAD

**Discussion by Gimno** [awawa-dev/HyperHDR#679] [Q&A]
Hello,
i have a 65" Tv and want to install HyperHDR.

What hardware i need for this?

Is it possible use a RPi zero wh or is it better to use a RPi 3+ or the Amlogic S905x3 like this X96 MAX Plus Smart TV Box?

Is it right to buy:

MS2130
sk6812 (how many leds per meter?)
SN74AHCT125N
LED power supply
ESP32-S2

Is this all right?

**Comment by awawa-dev** [comment]
Hi

> Is it possible use a RPi zero wh or is it better to use a RPi 3+ or the Amlogic S905x3 like this X96 MAX Plus Smart TV Box?

RPi Zero 2W (4 cores) is fine, archaic Zero wh with just one core is not recommended, but it should work for not demending configuration somehow.

Amlogic S905x3 boxes may differ significantly depending on the manufacturer, revision and Wifi chip used. They have been worth considering over the last 2 years due to the hopeless (well, profitable for them) Rapsberry Pi foundation policy and lack in the market. Now you can consider other alternatives ex. the brilliant mini intel n100 platform, which offers several times more performance (and if we compare accelerated codecs for media center consideration then RPi lags even further...even latest RPi5) for similar money. Or Raspberry Pi 4 or 5 since they are already easily available: usb3.0 is important especially for the ms2130 USB3.0 grabber.

> ESP32-S2 + SN74AHCT125N

Ok, but suggest a bit newer and simpler solution: one of recommended Adafruit/Pimoroni Pico board + level shitfer all-in-one board: https://github.com/awawa-dev/HyperHDR/discussions/561 . Or just generic Pico(rp2040) + SN74AHCT125N.

> sk6812 (how many leds per meter?)

well you have to measure it because it depends on the model, for example if it has a large base that you don't have to put the LEDs behind. If the LEDs will be very close to the wall then you can use 96 or 60 LEDs/meter. If further, consider using 30 LEDs to avoid mixing light from neighboring LEDs but this also means lower brightness. Also that depends on the power supply you plan to use: 60 and 96 are more demanding.

**Comment by dinjojo** [comment]
I have similar query for hardware here is my current setup

Esp32 installed with wled
WS2812B 5meter
Hyperion running on Ubuntu server on x86
Hyperion ambilight android app to capture screen

I'm happy with my setup but want to get it working with drm based apps had below queries

1. Do I need rpi of any sort? I can run hyperhdr on my x86 machine.
2. Does hyper hdr works with any android screen capture app?
3. If rpi is a mandate then what is the easiest device which has least problems as I'm not that hardware cognizant to install level shifter

**Reply by awawa-dev** [reply]
> 1 Do I need rpi of any sort? I can run hyperhdr on my x86 machine.

The x86 architecture is well supported. In fact, HyperHDR is developed using the x64 platform (mainly Windows 10 and Ubuntu LTS).

> 2 Does hyper hdr works with any android screen capture app?

If it uses proto/flatbuffer then it should work. Also HDR tone mapping should work for these sources.

> 3 If rpi is a mandate then what is the easiest device which has least problems as I'm not that hardware cognizant to install level shifter

Are you not using a 3.3v to 5v level shifter in your current esp32/wled setup? You should. Or use the alternative HyperSerialPico platform: some Adafruit/Piromani boards have a built-in level shifter. But anyway you can connect to your current wled from HyperHDR running on RPi or on PC.

**Comment by dinjojo** [comment]
Thanks for quickly reply no I'm not using any level shifter.

I didn't understood whether a usb capture is must for hyperhdr

What is the port for protobuf for hyper hdr I'll see whether it works with hyper hdr


Do we have any discord

**Reply by awawa-dev** [reply]
> What is the port for protobuf for hyper hdr I'll see whether it works with hyper hdr

it's configurable in the network settings

> Do we have any discord

no

**Comment by dinjojo** [comment]
Thanks its working however i dont see any difference when HDR is enabled from hyper hdr 

Can subtitle light be suppressed ?

**Reply by awawa-dev** [reply]
> Thanks its working however i dont see any difference when HDR is enabled from hyper hdr

and did you enable it first correctly as it's instructed in protobuf info?

**Comment by dinjojo** [comment]
I did probably its more active when captured using USB device

**Reply by awawa-dev** [reply]
it works the same way for USB grabbers and protobuffer. Can be calibrated for better results but if you dont see any difference then something is misconfigurated in your setup.

**Reply by dinjojo** [reply]
I see it now not sure why the settings are not permanent


Is there any difference if captured via protobuf on app vs usb capture via rpi?

Any recommended settings?

Please also comment for what can be done for subtitle lights

**Comment by pivoivo** [comment]
Hey there. Just thinking about realising this "little" project as well. Thanks for the great work!!

I am still a bit confused about the parts. Mainly the Splitter and the Grabber.

So I was thinking to buy the following compontens:

HyperHDR Host: RasPi 4
Controller: PiZero 2040 + Leveler SN74AHCT125N or does the Scorpio Version has any benefts? Was just wondering since it costs 3x the price of Pico + leveler
LEDs: BTF-LIGHTING RGBW RGBNW SK6812 Naturweiß Weiß 60LEDs/m or is more/less better? [LEDs](https://www.amazon.de/BTF-LIGHTING-Nat%C3%BCrliches-Individuell-adressierbar-nicht-wasserdicht/dp/B01N5ATQZT)
Regarding future Updates (77inch TV) I was thinking about just leaving the whole LED Strip together. Is it possible to just bend the strip instead of cutting and soldering it? Or is that not recommended? If it is possible and I read it right it would be possible to just disable the unused LEDs at the end, right and once upgraded to an 77inch TV reanable them, right? I want to solder as little as I have to, since I have never done it before. Does Hue Gradient Strips would have any benefits over the SK6812?
PSU: [LRS 150F-5](https://www.reichelt.de/schaltnetzteil-geschlossen-110-w-5-v-22-a-mw-lrs-150f-5-p202979.html)
Since with an Update I would need to use all of the 300 LEDs within 5m the WLED calculator says I would need a 65.70 W / 13.14 A PSU. Or is there any better alternative? Or is the overhead jsut right? Would a I guess a 100F-5 would also work then?



Grabber: [Grabber](https://de.aliexpress.com/item/1005004912019386.html) or do you have other Options/Links?

Splitter: I was wondering if it is worth to buy a 2.1 Splitter just in case for future Updates would this one work? [HDMI 2.1 Splitter](https://www.amazon.de/Splitter-HDCP2-3-Vision-Scaler-Monitore/dp/B0C739QQ4Q/ref=sr_1_3?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=250XWVT64C0EY&dib=eyJ2IjoiMSJ9.rkG9-USkgcmoOC1uSxDEgKjFrSlicc9nFk6SORcTOz4j0v_182ydvVVIsuVE2m72SAXoS2_vsz3FokluB_7byAC5xJwUDR2c3DhzT6dQMFsR5I_u9NALLoasVo8altZizJ-3a7deL1Gx_SOYakFDMBFXGG5Nrc6o-_qEtS76WG_02CsW-G_n0IdDECDYNLiuY-S1uK6PyW3AGNlNbcATWYgVFdl-E9EZzfBmU_a6MnY.rQQ8zdrGU7YUeQ8GfZ7qJbfyWDIYLeOooesZbsEeL7k&dib_tag=se&keywords=hdmi+2.1+splitter+scale&qid=1709891308&sprefix=hdmi+2.1+splitter+scal%2Caps%2C145&sr=8-3)

or should I better go with: [HDMI 2.0 Splitter](https://www.amazon.de/Splitter-HDCP2-2-18Gbps-Vision-Monitors-Grau-Schwarz/dp/B083JVPXDY/ref=sr_1_1?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=2RP6YWPZQ1CWI&dib=eyJ2IjoiMSJ9.ttibecTQ70kNCZdUUBpKLgdAbwFfdQFgDryBO7YdavmHkIst3aBHvtKe-JJSN6RjHZMSZ6bK7NxI89R6Rm2FU04Jjx5aCXhknM62gFMtmCsG_RIdQVCU5qjfQuPl9uqFke20kOn6zRSfXERM_jD697O7Q3RO4WW6B2mIKIyppJiRquBE5wW3aUZYSDfbxAB9gMhT-Ke3Xf6t6gCuWNw16AeOKbdmivHjwLYA1kz8m6U.by7HutbAOqlXQtum8XhlPrBc4GMqmP1fAB-R1bnHezY&dib_tag=se&keywords=hdmi+scaler+1in+2out&qid=1709892218&sprefix=hdmi+scaler+1in2ou%2Caps%2C161&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1)

Where will I need powersupplies? One for the Pi4 (dedicated), one for the HDMI Splitter (dedicated) and one for the LEDs (Mean-Well)?  The Pizero will get power via USB, right?

And for the cabeling it would be like this, then?

NvidiaShield ---HDMI Cable---> AVR HDMI in
AVR ---HDMI Cable --> HDMI Splitter in Port
HDMI Splitter out Port 1 (pass through)---HDMI Cable---> TV
HDMI Splitter out Port2 (downscale) ---HDMI Cable ---> Grabber
Grabber ---USB--> Rpi4
Rpi4 ---USB Cable---> PiZero
PiZero ---soldering--> LEDs
LEDs --soldering--> PSU

Is that correct? 

Hope I kind of understood how things work. 
Kind regards :)

**Comment by awawa-dev** [comment]
@pivoivo 

If you want to go with sk6812RGBW **neutral** it's your choice, my opinion is well known but it's your money and setup.

If you bend the tapes, you reduce the cross-section of the paths/increase the resistance. This is also why power cables should not be bent to a very small radius. Besides, you will definitely have problems with uniform backlight. Test soldering on some small, unnecessary pieces. There are plenty of tutorials on YouTube. It's not a matter of innate skills, but simple, repeatable patterns and appropriate equipment (soldering iron, tin, flux, people often forget about the latter, but it makes soldering much easier).

If you don't mind soldering, it's probably not worth paying for ready-made solutions from Pico, although when you add up everything, it won't be as good as you write. Besides, if you don't need a multi-segment, Adafruit ItsyBitsy for 2/3 of the price of Scorpio is enough.

I have this older/cheaper Ezcoo. It works OK, although I have been using Elgato with passthrou instead for a long time.

If you use AWA drivers (HyperSerial/HyperSpi) then 13A for 300 sk6812 RGBW is definitely overshoot. A solid 10A power supply is enough as long as you provide it with adequate ventilation (even passive ventilation without blocking air circulation). Why? Because the greatest load will be when the LED strip starts displaying white at 100%. But in AWA for RGBW both channels never work at 100%. It depends on the calibration, but let's say you can have, for example: 100% of the white channel and 0% for the RGB channels, or on the other hand, 0% of the white channel and 100% for the RGB channels. You will never have a case of 100% white channel and 100% RGB channel, which corresponds to the highest load.

**Comment by pivoivo** [comment]
@awawa-dev Thanks a lot for the clarification and response.
Thanks a lot for pointing out the LEDs I wanted to go with cold white but linked the wrong ones. These would the preferred ones.[LEDs](https://www.amazon.de/BTF-LIGHTING-Nat%C3%BCrliches-Individuell-adressierbar-nicht-wasserdicht/dp/B01N49DSC1?th=1)

If I cut tapes and later update to a bigger screen I guess I would need multi-segment? So going with pico scorpio or self-made solution would be preferred, I guess. An what do you mean with "although when you add up everything, it won't be as good as you write". I do not know that expression. 

What do you meant with "work OK"? Doing the job would be enough for me. What are the benefits of Elgato?

Even with a 77 inch Screen I would not use a lot more then the 5m LEDs which would sum up to 300 in total. Or are there any strips that have more then 60/m which are recommended? Otherwise the mentioned PSU from you would be fince for all of my usecases?

**Comment by awawa-dev** [comment]
> What do you meant with "work OK"? Doing the job would be enough for me. What are the benefits of Elgato?

I just haven't noticed any problems with Ezcoo. It works correctly and even forces LLDV with Dolby Vision, which not every splitter can do (actually, I don't know if any apart from Ezcoo or HDFury splitters). If you need HDCP then you can forget about Elgato (at least not without a HDMI splitter) but in my setup HDCP is disabled on PC.

> An what do you mean with "although when you add up everything, it won't be as good as you write". 

I meant the price compared to DIY solution.

> If I cut tapes and later update to a bigger screen I guess I would need multi-segment? 

it will work even without multi-segment. The question is whether the refresh-rate will be high enough for you to be satisfied with it. Statistics are provided so you know what to expect.

> Even with a 77 inch Screen I would not use a lot more then the 5m LEDs which would sum up to 300 in total. Or are there any strips that have more then 60/m which are recommended? Otherwise the mentioned PSU from you would be fince for all of my usecases?

https://github.com/awawa-dev/HyperSerialPico/discussions/20#discussioncomment-8703651

**Comment by pivoivo** [comment]
Ok. Thanks a lot for that Aliexpress link. But with all that new options I am confused again.

To be save for future updated I consider buying the ezcoo HDMI 2.1 Splitter and then the Adafruit Feather RP2040 Scorpio. The more LEDs the better? So considering buying the 144LEDs/m or 90/m. Will I even notice a diference compared to the 60/m?
Just to be clear. I would need to buy  SK6812 RGBW not the SK6812 RGBNW, right?

I do not really know what to do with the refresh-rate specs. I can not tell what is needed/good/ok.  If better, why should go for multi segment from the beginning or is there anything else to worry about then? I will not need more then the 55hz i guess? Since watching movies will not require such high hz and for gaming I am usually fine with maxed out 60hz. I should also mentions, that the tv is quite close to the wall.

**Reply by awawa-dev** [reply]
The new ezcoo costs nearly 3 times as much as the older one. The question is do you need these new features? Besides, I haven't tested it, and it is not known whether something new for less money will appear in the near future, because HDMI2.1-compatible devices are constantly (albeit slowly) cheaper.

Movies are generally only about 24Hz, but on LEDs you need more to ensure a smooth transition between frames (if you have smoothing enabled in HyperHDR). This is an individual matter, but it seems that it is not worth going below 50-60Hz.
